js/ui/control/fullscreen_control.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const DOM = require('../../util/dom');
+const util = require('../../util/util');
+const window = require('../../util/window');
+
+/**
+ * A `FullscreenControl` control contains a button for toggling the map in and out of fullscreen mode.
+ *
+ * @implements {IControl}
+ * @example
+ * map.addControl(new mapboxgl.FullscreenControl());
+ * @see [View a fullscreen map](https://www.mapbox.com/mapbox-gl-js/example/fullscreen/)
+ */
+
+class FullscreenControl {
+
+    constructor() {
+        this._fullscreen = false;
+        util.bindAll([
+            '_isFullscreen',
+            '_onClickFullscreen',
+            '_onKeyDown',
```

Comment:
Also I don't think you need to bind `_isFullscreen` bc its not used as an event listener.